File: src/languages/csharp/fixtures/Services.cs
namespace MyApp
{
    public class Calculator
    {
        public int Add(int a, int b)
        {
            return a + b;
        }

        public int Multiply(int a, int b)
        {
            return a * b;
        }
    }

    public class UserRepository : IRepository<User>
    {
        public User GetById(int id)
        {
            return new User("Unknown", 0);
        }

        public void Save(User entity)
        {
            // Save logic
        }
    }

    public struct Point
    {
        public int X;
        public int Y;
    }

    public delegate void EventCallback(string message);
}

File: src/languages/php/fixtures/Services.php
<?php

namespace MyApp\Services;

use MyApp\Models\User;
use MyApp\Models\Repository;

class Calculator
{
    public function add(int $a, int $b): int
    {
        return $a + $b;
    }

    public function multiply(int $a, int $b): int
    {
        return $a * $b;
    }
}

class UserRepository implements Repository
{
    public function find(int $id): ?User
    {
        return new User("Unknown", 0);
    }

    public function save(mixed $entity): void
    {
        // Save logic
    }
}

const DEFAULT_TIMEOUT = 30;
const MAX_RETRIES = 3;

function helper_function(): string
{
    return "helper";
}
